torch_np/_detail/_reductions.py

```diff
@@ -161,3 +166,97 @@ def var(tensor, axis=None, dtype=None, ddof=0, *, where=NoValue):
     result = tensor.var(dim=axis, correction=ddof)
 
     return result
+
+
+# cumsum / cumprod are almost reductions:
+#   1. no keepdims
+#   2. axis=None ravels (cf concatenate)
+
+
+def cumprod(tensor, axis=None, dtype=None):
+    if dtype == torch.bool:
+        dtype = _scalar_types.default_int_type.dtype
+    if dtype is None:
+        dtype = tensor.dtype
+
+    result = tensor.cumprod(axis=axis, dtype=dtype)
```

Comment:
`axis=None` is misleading here, thanks! This function never gets a None for axis.
 The way it works currently (there will be a cleaning pass over) is:
 `ndarray.cumprod` is a decorated by `_decorators.axis_none_ravel_wrapper` which eventually calls https://github.com/Quansight-Labs/numpy_pytorch_interop/blob/main/torch_np/_detail/_util.py#L121 which does
`if axis is None: return tensor.ravel(), 0`

Have removed the defaults for the axis argument to avoid confusing a reader.